xdem/spstats.py

```diff
@@ -91,6 +90,73 @@ def random_subset(dh: np.ndarray, coords: np.ndarray, nsamp: int):
 
     return dh_sub, coords_sub
 
+def create_circular_mask(h: float, w: float, center: Optional[list[float]] = None, radius: Optional[float] = None) -> np.ndarray:
+    """
+    Create circular mask on a raster
+
+    :param h: height of array
+    :param w: width of array
+    :param center: center
+    :param radius: radius
+    :return:
+    """
+
+    if center is None:  # use the middle of the image
+        center = [int(w / 2), int(h / 2)]
+    if radius is None:  # use the smallest distance between the center and image walls
+        radius = min(center[0], center[1], w - center[0], h - center[1])
+
+    Y, X = np.ogrid[:h, :w]
+    dist_from_center = np.sqrt((X - center[0]) ** 2 + (Y - center[1]) ** 2)
+
+    mask = dist_from_center <= radius
+
+    return mask
+
+def create_ring_mask(h: float, w: float, center: Optional[list[float]] = None, in_radius: float = 0., out_radius: float = 0.) -> np.ndarray:
+    """
+    Create ring mask on a raster
+
+    :param h: height of array
+    :param w: width of array
+    :param center: center
+    :param in_radius: inside radius
+    :param out_radius: outside radius
+    :return:
+    """
+
+    mask_inside = create_circular_mask(h,w,center=center,radius=in_radius)
+    mask_outside = create_circular_mask(h,w,center=center,radius=out_radius)
```

Comment:
By default, the in_radius and out_radius are equal, so will it not return an empty mask?? I would avoid this as it can be confusing.

Reply:
Now the defaults return a full circle with `inside_radius` of 0